### PR TITLE
fixed core.login trigger

### PR DESCRIPTION
**Pull request by billybennett**
In reference to issue #371

### CODE REVIEW

**Comment by benlangfeld** [comment]
It's good practice to include the detail from your issue in the commit message. Could you do that?

File: candy.bundle.js [hunk 1]
```diff
@@ -2027,19 +2027,22 @@ Candy.Core.Event = function(self, Strophe, $) {
...
-	 *   candy:core.login using {presetJid}
+	 *   candy:core.login using {messageKey, presetJid}
 	 */
-    self.Login = function(presetJid) {
+    self.Login = function(messageKey, presetJid) {
         /** Event: candy:core.login
 		 * Triggered when the login window should be displayed
 		 *
 		 * Parameters:
+         *   (String) messageKey - i18n message key
```
Comment (benlangfeld) [inline candy.bundle.js]: Indenting

File: candy.bundle.js [hunk 2]
```diff
@@ -241,10 +241,10 @@ Candy.Core = function(self, Strophe, $) {
             _connection.connect(_getEscapedJidFromJid(jidOrHost) + "/" + _options.resource, null, Candy.Core.Event.Strophe.Connect);
             _user = new self.ChatUser(null, nick);
         } else if (jidOrHost) {
-            Candy.Core.Event.Login(jidOrHost);
+            Candy.Core.Event.Login(null, jidOrHost);
         } else {
             // display login modal
-            Candy.Core.Event.Login();
+            Candy.Core.Event.Login(null, null);
```
Comment (benlangfeld) [inline candy.bundle.js]: It's not actually necessary to pass these options - they'll come out as `undefined` if you leave them off the method invocation.

File: candy.bundle.js [hunk 3]
```diff
@@ -2027,19 +2027,22 @@ Candy.Core.Event = function(self, Strophe, $) {
 	 * Notify view that the login window should be displayed
 	 *
 	 * Parameters:
-	 *   (String) presetJid - Preset user JID
+	 *   (String) messageKey - i18n Message Key or null
```
Comment (benlangfeld) [inline candy.bundle.js]: Except you're not actually passing the key, you're passing the full translated string.

File: candy.bundle.js [hunk 4]
```diff
@@ -3035,10 +3038,13 @@ Candy.View.Observer = function(self, $) {
...
 	 * Parameters:
 	 *   (jQuery.Event) event - jQuery Event object
-	 *   (Object) args - {presetJid}
+	 *   (Object) args - {messageKey, presetJid}
 	 */
     self.Login = function(event, args) {
-        Candy.View.Pane.Chat.Modal.showLoginForm(null, args.presetJid);
+        args = args || {};
+        args.messageKey = args.messageKey || null;
+        args.presetJid = args.presetJid || null;
```
Comment (benlangfeld) [inline candy.bundle.js]: This null stuff is not necessary. Use `undefined` implicitly to help you.

**Comment by benlangfeld** [comment]
Please rebase this against the `dev` branch. That is where effort is currently going.

File: candy.bundle.js (hunk 1 above)
Comment (billybennett) [inline candy.bundle.js]: The indenting shown here is due to tabs being used in the source (not mine i ripped the tab key out of my keyboard years ago:

keyboard

Comment (benlangfeld) [inline candy.bundle.js]: Yep. Candy uses tabs. Please keep your submissions consistent.

**Pull request closed without merging**